tools/read-dependencies

```diff
@@ -48,6 +48,11 @@ MAYBE_RELIABLE_YUM_INSTALL = [
     done
     error ":: running yum install --cacheonly --assumeyes $*"
     yum install --cacheonly --assumeyes "$@"
+    if grep -q "^proxy=" /etc/yum.conf; then
```

Comment:
this is too late to do this, isn't it?

MAYBE_RELIABLE_YUM_INSTALL does the "download" up above (in a while loop), and then the "install" with `--cacheonly` below.  I think you want the change to the mirror list to be done before the download above.

Reply:
It is not too late: the mirror list was already mangled by the `run-container` script, in the `wait_for_boot()`. I touch the relevant lines in this same commit. Here we download and install new packages, and after doing so the URLs get mangled, just in case the new packages added new repos.

Reply:
maybe you shoudl make it do that before and after then?  then you would not be depending on run-container doing it.

Reply:
Agreed

Reply:
@smoser done